roofit/roofitcore/src/RooDataHist.cxx

```diff
@@ -1128,51 +1128,58 @@ Double_t RooDataHist::weight(const RooArgSet& bin, Int_t intOrder, Bool_t correc
   // Handle all interpolation cases
   _vars.assignValueOnly(bin) ;
 
-  auto varInfo = getVarInfo();
+  return weightInterpolated(_vars, intOrder, correctForBinSize, cdfBoundaries);
+}
+
+
+double RooDataHist::weightInterpolated(const RooArgSet& bin, int intOrder, bool correctForBinSize, bool cdfBoundaries) {
+  VarInfo const& varInfo = getVarInfo();
 
-  Double_t wInt(0) ;
+  auto centralIdx = calcTreeIndex(bin, true);
```

Comment:
That's very optional, but you could make temporary stuff `const`. You don't inadvertently overwrite it and make sure that it gets optimised out. (It will of course also be optimised out when it's not `const`, unless you write to it by accident.)

Reply:
Sure, that's a good thing to do!